### ISSUE TITLE
Review: Landing.jsx and Audited.jsx

### ISSUE THREAD

**Issue by Sydnee-You** [manoa-inspire/MATP#99]
## Overview

The focus for this code review will be centered around the Landing and Audited pages.    
Please pay attention too:
* Javascript issues
* React components

## Review Branch

[review-4](https://github.com/manoa-inspire/MATP/tree/review-4)

## Files to review

* [Landing.jsx](https://github.com/manoa-inspire/MATP/blob/review-4/app/imports/ui/pages/Landing.jsx)
* [Audited.jsx](https://github.com/manoa-inspire/MATP/blob/review-4/app/imports/ui/pages/Audited.jsx)

## Checklists

* [Architecture](../../checklists/architecture-checklist.md)
* [Design](../../checklists/design-checklist.md)
* [JS](../../checklists/js-checklist.md)
* [React](../../checklists/react-checklist.md)
* [Testing](../../checklists/testing-checklist.md)

## Due date

10/07/2024 at 9:00am

## For more information

The review process is documented at: http://courses.ics.hawaii.edu/ics414s21/morea/review/reading-idpm-review.html

**Comment by blakewatanabe** [comment]
## Landing.jsx

DE-01: Can you refactor - The component is straightforward, but maybe you could refactor the feature cards into a separate FeatureCard component to improve reusability.

## Audited.jsx

DE-01: Can you refactor - The tooltip functions could be refactored into a single reusable function that accepts the tooltip text as an argument to avoid redundancy.

DE-07: Ensure code is DRY- The tooltip elements are a bit repetitive. Maybe you could create a generic tooltip function to DRY up the code.

**Comment by cjochim** [comment]
Landing.jsx
- DE-01: Feature cards could be refactored to avoid repetition.
- DE-03: The commented code, the third carousel item, can be deleted. I was the one who commented out that code and it is no longer needed.

Audited.jsx
- DE-07: Even though each tooltip has a separate function, the code for the tooltips is repetitive because they follow the same structure. To avoid repetition, we can refactor into a reusable function.

**Comment by TannerBerry** [comment]
Landing.jsx
AR-01 (ln N/A): The file name landing.jsx should follow the lowercase naming convention and is correct.
DE-01 (ln 28): There is commented-out code (<Carousel.Item> with a third slide), which can be removed if not needed, or reintroduced once the third image is ready.
DE-06 (ln 9): The ESLint comment disabling unused variable checks (eslint-disable-next-line no-unused-vars) indicates that e is not being used in the handleSelect function. It might be better to remove the parameter e if it's not necessary.

Audited.jsx
AR-01 (ln N/A): The file name audited.jsx follows the correct lowercase naming convention and is correct.
ES-01 (ln 10, ln 14, ln 18, ln 22, ln 26, ln 30, ln 34, ln 38, ln 42, ln 46, ln 50, ln 54, ln 58, ln 62, ln 66, ln 70, ln 74, ln 78, ln 82, ln 86, ln 90, ln 94): Multiple instances of eslint-disable-next-line react/jsx-props-no-spreading are used to suppress warnings. It is better to refactor the code to avoid spreading props unless absolutely necessary.
DE-06 (ln N/A): Code is generally readable, though the repeated use of OverlayTrigger components for tooltips might benefit from abstraction into a reusable component to reduce redundancy.

**Comment by Sydnee-You** [comment]
**Landing.jsx**

DE-03 on lines 41 to 42 (unused code)
ES-01 on line 12 (unused parameter e)


**Audited.jsx**

DE-01 in the beginning (not sure how to reduce
the duplication but it might be possible)

DE-05 line 92 (what does this mean?)

ES-01 line 1 (a lot is similar AuditedBalanceInput.jsx)

**Comment by RomaMalasarte** [comment]
**Landing.jsx** 
L40-42: Design-03, Remove code that is being unused 

L11: Design-04,  e parameter is unused, remove unecessary comments

**Audited.jsx** 
L105-112 and alike: Design-01: refactor the code that is similar to this format to avoid redundancy

L107-109: Design-08: Ensure code is DRY. There is repetition with the Form.Group as mentioned previously as well as OverlayTrigger

**Comment by juncell-venzon** [comment]
Landing.jsx
DE-01:  The same ImageCarousel is used twice. Add different images or remove the second one if it’s not needed.
DE-03: The cards are repetitive. Create a FeatureCard component to reuse the structure.
DE-04: Add "Learn More" buttons or links to the cards for more details.

Audit.jsx
DE-02: The buttons at the bottom are hardcoded with inline styles. It would be better to move the styles to CSS classes for consistency.
RE-01: The form fields and tooltips are repeated. Extract them into a separate form component to reduce duplication.

**Comment by luisy-eng** [comment]
### Review of `Landing.jsx`

#### **1. Architecture Review:**
1. **AR-01:** *Directory Structure* - Check if the file follows a proper directory structure.
   - **Observation:** The file seems to be located properly within the project’s structure. Ensure that components are placed in a directory that reflects their role in the application.
   
2. **AR-02:** *File Naming Convention* - Ensure the file name reflects the component's purpose.
   - **Observation:** The name `Landing.jsx` is appropriate for a landing page component.

3. **AR-04:** *Component Organization* - Components should be modular and organized.
   - **Observation:** The component could be modularized further, breaking down large sections into smaller reusable components.

#### **2. Design Review:**
1. **DE-03:** *Use of Layout* - Properly use layout grids and ensure responsiveness.
   - **Observation:** The layout of the `Landing.jsx` page is responsive, but could benefit from further testing across multiple screen sizes.

2. **DE-06:** *Component Reusability* - Look for repetition that could be abstracted into reusable components.
   - **Observation:** There are several repeated JSX elements, which could be extracted into reusable components to improve maintainability.

#### **3. JavaScript Review:**
1. **JS-02:** *Variable Declarations* - Ensure the use of `const` and `let` appropriately.
   - **Observation:** Proper usage of `const` for constants and `let` for variables is followed.

2. **JS-07:** *DRY Principle* - Avoid repetitive code.
   - **Observation:** Repeated JSX sections are present, and breaking these into functions or components would adhere better to the DRY principle.

#### **4. React Review:**
1. **RE-01:** *Functional vs Class Components* - Prefer functional components for simplicity.
   - **Observation:** If this is not a class-based component, it follows the recommendation to use functional components.

2. **RE-03:** *Prop Drilling* - Minimize prop drilling by using context or other state management tools.
   - **Observation:** If there is deep prop drilling, it should be refactored to use context API or a state management solution like Redux.

---

### Review of `Audited.jsx`

#### **1. Architecture Review:**
1. **AR-01:** *Directory Structure* - Check if the file follows a proper directory structure.
   - **Observation:** Similar to `Landing.jsx`, the file seems appropriately placed but ensure consistency in how the component hierarchy is structured.

2. **AR-02:** *File Naming Convention* - Ensure the file name reflects the component's purpose.
   - **Observation:** The name `Audited.jsx` suggests it is related to audited data or elements, which fits well if that's its intended purpose.

#### **2. Design Review:**
1. **DE-03:** *Use of Layout* - Properly use layout grids and ensure responsiveness.
   - **Observation:** The layout looks solid, but there is room to ensure all components are responsive, especially on mobile.

2. **DE-06:** *Component Reusability* - Look for repetition that could be abstracted into reusable components.
   - **Observation:** There are areas where form groups and inputs are repeated, suggesting these could be moved into a more generic form field component for reuse.

#### **3. JavaScript Review:**
1. **JS-02:** *Variable Declarations* - Ensure the use of `const` and `let` appropriately.
   - **Observation:** Proper use of `const` and `let` is observed throughout.

2. **JS-07:** *DRY Principle* - Avoid repetitive code.
   - **Observation:** Like in `Landing.jsx`, there are sections of repeated form inputs that could be abstracted into reusable components.

#### **4. React Review:**
1. **RE-01:** *Functional vs Class Components* - Prefer functional components for simplicity.
   - **Observation:** If it is a functional component, then this guideline is followed well. Otherwise, consider converting any class-based logic to functional components.

2. **RE-09:** *Stateless Components* - Ensure that components are stateless if they do not manage their own state.
   - **Observation:** If this component doesn't manage state, it should be stateless, relying on props for data.

**Comment by cash-baker** [comment]
## Landing.jsx

DE-01:  Feature cards should be refactored to improve reusability and code readability.

## Audited.jsx

DE-06: Repetitive elements of code, namely the tooltips. Should be refactored into reusable components
